unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/ControlsModule/SettingsControlView.cs

```diff
@@ -6,20 +7,145 @@ namespace DCL.SettingsPanelHUD.Controls
     public interface ISettingsControlView
     {
         void Initialize(SettingsControlModel controlConfig, SettingsControlController settingsControlController);
+        void RefreshControl();
     }
 
-    public class SettingsControlView : MonoBehaviour, ISettingsControlView
+    public abstract class SettingsControlView : MonoBehaviour, ISettingsControlView
     {
         [SerializeField] private TextMeshProUGUI title;
+        [SerializeField] private GameObject betaIndicator;
+        [SerializeField] private CanvasGroup canvasGroup;
 
         protected SettingsControlController settingsControlController;
+        protected bool skipPostApplySettings = false;
+
+        private SettingsControlModel controlConfig;
+
+        private void OnEnable()
+        {
+            if (settingsControlController == null)
+                return;
+
+            skipPostApplySettings = true;
+            RefreshControl();
+        }
 
         public virtual void Initialize(SettingsControlModel controlConfig, SettingsControlController settingsControlController)
         {
+            this.controlConfig = controlConfig;
             this.settingsControlController = settingsControlController;
-            this.settingsControlController.Initialize();
+            this.settingsControlController.Initialize(this);
+            title.text = controlConfig.title;
+            betaIndicator.SetActive(controlConfig.isBeta);
+
+            foreach (BooleanVariable flag in controlConfig.flagsThatDisablesMe)
+            {
+                flag.OnChange += OnAnyDisableFlagChange;
+                OnAnyDisableFlagChange(flag.Get(), false);
```

Comment:
@pbosio this is an example of a good use of your `Ensure` pattern with your promise class. Maybe we can integrate this pattern in `BaseVariable` in the future to avoid having to call the event manually after suscription?

NOTE: Don't take this as specific feedback, just discussion.

Reply:
yeah, we should start that discussion at some point. Not entirely sure how that promise would help in this scenario but I'm really curious about what are you thinking :)